### DISCUSSION TITLE
ext_foreign_toplevel_handle_v1::identifier support?

### DISCUSSION THREAD

**Discussion by davidphilipbarr** [labwc/labwc#1924] [Ideas]
I'm not completely sure what i'm asking here, i've been looking at ways to get unique identifiers from the command line for windows, lswt seems to support identifiers, which i think is this ext_foreign_toplevel_handle_v1::identifier - I might be misunderstanding the purpose, but it seems like it would provide this function?  Am i right or just confused? labwc doesn't seem to have this implemented, i'm not sure :)

**Comment by Consolatis** [comment]
There is the [ext-foreign-toplevel-list](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/staging/ext-foreign-toplevel-list/ext-foreign-toplevel-list-v1.xml) protocol and the [wlr-foreign-toplevel-management](https://gitlab.freedesktop.org/wlroots/wlr-protocols/-/blob/master/unstable/wlr-foreign-toplevel-management-unstable-v1.xml) protocol.

ext-foreign-toplevel-list is basically useless at this point because it is designed to mainly provide a stable identifier and only serve as the base protocol for others that can then reference the handles (like [ext-foreign-toplevel-state](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/196), [ext-screencopy](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/124) or [ext-workspaces](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/40) which are not yet accepted). The wlr-foreign-toplevel protocol in contrast is allowing window listing and control, however it does not provide a stable identifier across wayland connections. There is no connection between the two protocols.

Labwc only implements the wlr-foreign-toplevel-management protocol, the ext-foreign-toplevel-list protocols is only available in the wlroots master branch which didn't yet have a release so labwc is not tracking it.

For the moment the only two options are
- using the app_id + title combination but the combination is neither required to be unique nor stable
- keep a wayland connection open, during the lifetime of the wayland connection the handles (object ids in wayland terminology) are required to be stable

**Reply by davidphilipbarr** [reply]
ok gotcha, thanks for the explanation!
